Sources/CasePaths/EnumReflection.swift

```diff
@@ -67,89 +67,48 @@ public func extract<Root, Value>(case embed: @escaping (Value) -> Root, from roo
     let metadata = EnumMetadata(assumingEnum: Root.self)
     let rootTag = metadata.tag(of: root)
 
-    if let cachedTag = cachedTag {
+    if let cachedTag = cachedTag, let cachedStrategy = cachedStrategy {
       guard rootTag == cachedTag else { return nil }
       return cachedStrategy.extract(from: root, tag: rootTag)
     }
 
-    guard
-      let rootStrategy = Strategy<Root, Value>(tag: rootTag),
-      let value = rootStrategy.extract(from: root, tag: rootTag)
-    else {
-      return nil
-    }
+    guard let value = Strategy<Root, Value>(tag: rootTag).extract(from: root, tag: rootTag)
+    else { return nil }
 
-    let embeddedValue = embed(value)
-    let embedTag = metadata.tag(of: embeddedValue)
+    let embedTag = metadata.tag(of: embed(value))
     cachedTag = embedTag
+    cachedStrategy = Strategy<Root, Value>(tag: embedTag)
 
-    guard cachedTag == rootTag else {
-      // I have the correct tag now, so I can now pick the correct strategy, even if I won't use it this time.
-      if let embedStrategy = Strategy<Root, Value>(tag: embedTag) {
-        cachedStrategy = embedStrategy
-      } else {
-        #if DEBUG
-          print(
-            "\(#function) can never extract values from a \(Root.self) case like \(embeddedValue) because I don't know how to convert its associated value to \(Value.self)!"
-          )
-        #endif
-        cachedStrategy = .unimplemented
-      }
+    guard cachedTag == rootTag else { return nil }
 
-      return nil
-    }
-
-    cachedStrategy = rootStrategy
     return value
   }
 }
 
 // MARK: - Private Helpers
 
-/// A `Strategy` is a way of extracting an associated value from an enum case.
-enum Strategy<Enum, Value> {
-  case unimplemented
-
-  /// The case's associated type is a zero-size inhabited type, so it only has a single possible inhabitant, which I can synthesize.
-  case void
-
-  /// The case is layout-compatible with `Value`, after tag-stripping (aka projection).
+private enum Strategy<Enum, Value> {
   case direct
-
-  /// The case stores its associated value indirectly. The case payload is a pointer to a heap object. The heap object's payload is layout-compatible with `Value`.
+  case existential(extract: (Enum) -> Any?)
   case indirect
-
-  /// The case stores a protocol existential, either directly or indirectly. This strategy is used when the enum case's associated value type is a protocol existential and the `CasePath`'s `Value` is a type that conforms to the protocol (but is not itself the protocol existential).
-  case existential(get: (Enum) -> Any?)
+  case unimplemented
+  case void
 }
 
 extension Strategy {
-  /// Choose the appropriate `Strategy` to extract a `Value` from an `Enum` if that `Enum`'s case tag is `tag`. If `assumedAssociatedValueType` is nil, I'll look up the associated value type in the `Enum` metadata.
-  init?(tag: UInt32, assumedAssociatedValueType: Any.Type? = nil) {
+  init(tag: UInt32, assumedAssociatedValueType: Any.Type? = nil) {
```

Comment:
I made this unfailable, since it seems like we can always return `.unimplemented` when it fails.

Reply:
If it's no longer going to print a warning when `init` fails, then I see no reason for `init` to be failable.